internal/compiler/wasm/wasm.go

```diff
@@ -1068,16 +1079,25 @@ func (c *Compiler) compileCallStmt(stmt *ir.CallStmt, result *[]instruction.Inst
 		fn = name
 	}
 
-	if index, ok := c.funcs[fn]; ok {
-		return c.compileInternalCall(stmt, index, result)
-	}
+	if _, ok := builtinsPartial[stmt.Func]; !ok {
```

Comment:
I think we could simplify this by just adding a conditional into the `compileInternalCall` function that checks if the call is partial and prepends the two operands (built-in identifier and context) to the call.